pkg/git/repospec.go

```diff
@@ -142,28 +155,45 @@ func peelQuery(arg string) (string, string) {
 
 func parseHostSpec(n string) (string, string) {
 	var host string
+	// Start accumulating the host part.
 	for _, p := range []string{
 		// Order matters here.
 		"git::", "gh:", "ssh://", "https://", "http://",
 		"git@", "github.com:", "github.com/"} {
-		if strings.ToLower(n[:len(p)]) == p {
+		if len(p) < len(n) && strings.ToLower(n[:len(p)]) == p {
 			n = n[len(p):]
-			host = host + p
+			host += p
+		}
+	}
+	if host == "git@" {
+		i := strings.Index(n, "/")
+		if i > -1 {
+			host += n[:i+1]
+			n = n[i+1:]
```

Comment:
Does this branch aim to support repo uris as
```
git@github.com/kubernetes-sigs/kustomize.git
```

Reply:
That url without the trailing `.git` is covered.  But adding the `.git` suffix (in that form) isn't covered by the new or old tests.

FWIW, a user doesn't need to use it, and adding the `.git` is an odd syntax if one also wants to target a particular directory inside the repo.